<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>ECMpy_function &mdash; ecmpy 2.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
        <script src="_static/my_custom.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="get_ecGEM_onestop" href="get_ecGEM_onestop.html" />
    <link rel="prev" title="AutoPACMEN_function" href="AutoPACMEN_function.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html">
            
              <img src="_static/logo.png" class="logo" alt="Logo"/>
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="AutoPACMEN_function.html">AutoPACMEN_function</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">ECMpy_function</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.DL_kcat_mw_calculation"><code class="docutils literal notranslate"><span class="pre">DL_kcat_mw_calculation()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.Determine_suitable_ecGEM"><code class="docutils literal notranslate"><span class="pre">Determine_suitable_ecGEM()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.GENENAME_2_ACC_from_uniprot"><code class="docutils literal notranslate"><span class="pre">GENENAME_2_ACC_from_uniprot()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.adj_reaction_kcat_by_database"><code class="docutils literal notranslate"><span class="pre">adj_reaction_kcat_by_database()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.adj_trans_model2enz_model"><code class="docutils literal notranslate"><span class="pre">adj_trans_model2enz_model()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.calculate_f"><code class="docutils literal notranslate"><span class="pre">calculate_f()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.calculate_f_v2"><code class="docutils literal notranslate"><span class="pre">calculate_f_v2()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.calculate_yield"><code class="docutils literal notranslate"><span class="pre">calculate_yield()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.change_enz_model_by_enz_usage"><code class="docutils literal notranslate"><span class="pre">change_enz_model_by_enz_usage()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.change_reaction_kcat_by_database"><code class="docutils literal notranslate"><span class="pre">change_reaction_kcat_by_database()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.combine_reactions_simles_sequence"><code class="docutils literal notranslate"><span class="pre">combine_reactions_simles_sequence()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.convert_bigg_met_to_inchikey"><code class="docutils literal notranslate"><span class="pre">convert_bigg_met_to_inchikey()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.convert_inchikey_to_smiles"><code class="docutils literal notranslate"><span class="pre">convert_inchikey_to_smiles()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.convert_to_irreversible"><code class="docutils literal notranslate"><span class="pre">convert_to_irreversible()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.create_file"><code class="docutils literal notranslate"><span class="pre">create_file()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.create_trace"><code class="docutils literal notranslate"><span class="pre">create_trace()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.draw_3d_rbas"><code class="docutils literal notranslate"><span class="pre">draw_3d_rbas()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.draw_cdf_fig"><code class="docutils literal notranslate"><span class="pre">draw_cdf_fig()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.draw_overfolw_fig"><code class="docutils literal notranslate"><span class="pre">draw_overfolw_fig()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.draw_trade_off"><code class="docutils literal notranslate"><span class="pre">draw_trade_off()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.drawphpp"><code class="docutils literal notranslate"><span class="pre">drawphpp()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.extract_metabolite_id"><code class="docutils literal notranslate"><span class="pre">extract_metabolite_id()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.generate_DLKCAT_input"><code class="docutils literal notranslate"><span class="pre">generate_DLKCAT_input()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.generate_random_colors"><code class="docutils literal notranslate"><span class="pre">generate_random_colors()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_PhPP_data"><code class="docutils literal notranslate"><span class="pre">get_PhPP_data()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_currency_metabolites"><code class="docutils literal notranslate"><span class="pre">get_currency_metabolites()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_enz_foldchange"><code class="docutils literal notranslate"><span class="pre">get_enz_foldchange()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_enz_model_use_enz_usage_by_eckcat"><code class="docutils literal notranslate"><span class="pre">get_enz_model_use_enz_usage_by_eckcat()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_enzyme_constraint_model"><code class="docutils literal notranslate"><span class="pre">get_enzyme_constraint_model()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_enzyme_constraint_model_percent"><code class="docutils literal notranslate"><span class="pre">get_enzyme_constraint_model_percent()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_fluxes_detail_in_model"><code class="docutils literal notranslate"><span class="pre">get_fluxes_detail_in_model()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_gene_subunitDescription"><code class="docutils literal notranslate"><span class="pre">get_gene_subunitDescription()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_genes_and_gpr"><code class="docutils literal notranslate"><span class="pre">get_genes_and_gpr()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_met_bigg_id"><code class="docutils literal notranslate"><span class="pre">get_met_bigg_id()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_min_enzyme_cost"><code class="docutils literal notranslate"><span class="pre">get_min_enzyme_cost()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_model_protein_sequence_and_mass"><code class="docutils literal notranslate"><span class="pre">get_model_protein_sequence_and_mass()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_model_substrate_obj"><code class="docutils literal notranslate"><span class="pre">get_model_substrate_obj()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_protein_mass_mapping_from_local"><code class="docutils literal notranslate"><span class="pre">get_protein_mass_mapping_from_local()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_reaction_kcat_mw"><code class="docutils literal notranslate"><span class="pre">get_reaction_kcat_mw()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_reaction_kcatmw_onestop"><code class="docutils literal notranslate"><span class="pre">get_reaction_kcatmw_onestop()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_reaction_kcatmw_onestop_by_AutoPACMEN"><code class="docutils literal notranslate"><span class="pre">get_reaction_kcatmw_onestop_by_AutoPACMEN()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_reaction_kcatmw_onestop_by_DLKcat"><code class="docutils literal notranslate"><span class="pre">get_reaction_kcatmw_onestop_by_DLKcat()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_reaction_mw"><code class="docutils literal notranslate"><span class="pre">get_reaction_mw()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_subunit_number"><code class="docutils literal notranslate"><span class="pre">get_subunit_number()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.get_yield_cost_efficiency"><code class="docutils literal notranslate"><span class="pre">get_yield_cost_efficiency()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.isoenzyme_split"><code class="docutils literal notranslate"><span class="pre">isoenzyme_split()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.json_load"><code class="docutils literal notranslate"><span class="pre">json_load()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.json_write"><code class="docutils literal notranslate"><span class="pre">json_write()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.parse_sabio_rk_for_eclist"><code class="docutils literal notranslate"><span class="pre">parse_sabio_rk_for_eclist()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.preprocess_metabolites"><code class="docutils literal notranslate"><span class="pre">preprocess_metabolites()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.run_FSEOF"><code class="docutils literal notranslate"><span class="pre">run_FSEOF()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.split_substrate_to_match_gene"><code class="docutils literal notranslate"><span class="pre">split_substrate_to_match_gene()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.standardize_folder"><code class="docutils literal notranslate"><span class="pre">standardize_folder()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ECMpy_function.trans_model2enz_json_model_split_isoenzyme"><code class="docutils literal notranslate"><span class="pre">trans_model2enz_json_model_split_isoenzyme()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="get_ecGEM_onestop.html">get_ecGEM_onestop</a></li>
<li class="toctree-l1"><a class="reference internal" href="model.html">model</a></li>
<li class="toctree-l1"><a class="reference internal" href="prediction_for_input.html">prediction_for_input</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">ecmpy</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">ECMpy_function</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/ECMpy_function.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-ECMpy_function">
<span id="ecmpy-function"></span><h1>ECMpy_function<a class="headerlink" href="#module-ECMpy_function" title="Permalink to this heading"></a></h1>
<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.DL_kcat_mw_calculation">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">DL_kcat_mw_calculation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">DLouputdf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metdf</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.DL_kcat_mw_calculation" title="Permalink to this definition"></a></dt>
<dd><p>Make the kcat_mw input file for ecGEM construction.</p>
<p>Arguments:
* DLouputdf: pandas.DataFrame - DLinputdf from DLKCAT.
* metdf: pandas.DataFrame - DataFrame containing metabolite, reaction, gene, similes, prosequence, mass information.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.Determine_suitable_ecGEM">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">Determine_suitable_ecGEM</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bigg_met_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.Determine_suitable_ecGEM" title="Permalink to this definition"></a></dt>
<dd><p>Determine if the provided model is suitable for constructing an enzyme-constrained model.</p>
<p>Arguments:
* model_file: str - File path of the model in SBML or JSON format.
* bigg_met_file: str - File path of the Bigg metabolite data file in tab-separated format.</p>
<p>Returns:
* result: str or list - If the model is suitable, returns “Suitable for constructing enzyme-bound models.”</p>
<blockquote>
<div><p>If there are errors, returns a list of error messages.</p>
</div></blockquote>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.GENENAME_2_ACC_from_uniprot">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">GENENAME_2_ACC_from_uniprot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">query</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">outfile</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.GENENAME_2_ACC_from_uniprot" title="Permalink to this definition"></a></dt>
<dd><p>get amin acid sequence and mass by Uniprot id</p>
<section id="arguments">
<h2>Arguments<a class="headerlink" href="#arguments" title="Permalink to this heading"></a></h2>
<p>query: Uniprot ID list.
outfile:  The path of the ACC file that shall be written</p>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.adj_reaction_kcat_by_database">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">adj_reaction_kcat_by_database</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">json_model_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">select_reactionlist</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">need_change_reaction_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">changed_reaction_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">EC_max_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_kcat_mw</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.adj_reaction_kcat_by_database" title="Permalink to this definition"></a></dt>
<dd><p>Use the kcat in database to change reaction kcat in model</p>
<section id="id1">
<h2>Arguments<a class="headerlink" href="#id1" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>json_model_path: The file storing json model.</p></li>
<li><p>select_reactionlist: reaction list need to change.</p></li>
<li><p>kcat_database_combined_file: combined kcat file got from autoPACMEN.</p></li>
<li><p>reaction_kcat_mw_file: reaction kcat/mw file.</p></li>
<li><p>reaction_kapp_change_file: changed file stored reaction kcat/mw.</p></li>
</ul>
<dl class="field-list simple">
<dt class="field-odd">return<span class="colon">:</span></dt>
<dd class="field-odd"><p>a dataframe stored new reaction kcat/mw .</p>
</dd>
</dl>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.adj_trans_model2enz_model">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">adj_trans_model2enz_model</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_kcat_mw</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">f</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ptot</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sigma</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lowerbound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">upperbound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">json_output_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.adj_trans_model2enz_model" title="Permalink to this definition"></a></dt>
<dd><p>Tansform cobra model to json mode with  
enzyme concentration constraintat.</p>
<section id="id2">
<h2>Arguments<a class="headerlink" href="#id2" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>model_file:   The path of sbml model</p></li>
<li><p>reaction_kcat_mw_file: The path of storing kcat/MW value of the enzyme catalyzing each</p></li>
</ul>
<blockquote>
<div><p>reaction in the GEM model</p>
</div></blockquote>
<ul class="simple">
<li><p>f: The enzyme mass fraction</p></li>
<li><p>ptot: The total protein fraction in cell.</p></li>
<li><p>sigma: The approximated average saturation of enzyme.</p></li>
<li><p>lowerbound:  Lowerbound  of enzyme concentration constraint.</p></li>
<li><p>upperbound:  Upperbound  of enzyme concentration constraint.</p></li>
</ul>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.calculate_f">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">calculate_f</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_abundance_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_mw_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_mw_colname</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_abundance_colname</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.calculate_f" title="Permalink to this definition"></a></dt>
<dd><p>Calculate the enzyme mass fraction (f) based on protein abundance and molecular weight.</p>
<p>Arguments:
* model_file: str - File path of the SBML model.
* gene_abundance_file: str - File path of the gene abundance data file.
* gene_mw_file: str - File path of the gene molecular weight data file.
* gene_mw_colname: str - Name of the column in the gene molecular weight file containing the molecular weight values.
* gene_abundance_colname: str - Name of the column in the gene abundance file containing the abundance values.</p>
<p>Returns:
* f: float - The enzyme mass fraction.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.calculate_f_v2">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">calculate_f_v2</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_abundance_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_abundance_colname</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">taxonom_id</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.calculate_f_v2" title="Permalink to this definition"></a></dt>
<dd><p>Calculate the enzyme mass fraction (f) based on protein abundance.</p>
<p>Arguments:
* model_file: str - File path of the SBML model.
* gene_abundance_file: str - File path of the gene abundance data file.
* gene_abundance_colname: str - Name of the column in the gene abundance file containing the abundance values.
* taxonom_id: str - Taxonomy ID of the organism.</p>
<p>Returns:
* f: float - The enzyme mass fraction.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.calculate_yield">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">calculate_yield</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_substrate</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">substrate_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">glc_concentration</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">columns</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">correspond_rxn</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">yield_list</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.calculate_yield" title="Permalink to this definition"></a></dt>
<dd><p>Calculate yield using the provided model and glucose concentration.</p>
<p>Arguments:
* model: cobra.Model - COBRApy model object.
* use_substrate: str - substrate ID in model.
* substrate_name: str - substrate name.
* glc_concentration: float - substrate concentration.
* columns: list - List of reactions corresponding to the columns in yield_list
* yield_list: pd.DataFrame - DataFrame to store the calculated yields.</p>
<p>Returns:
* yield_list: pd.DataFrame - DataFrame with updated yield values.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.change_enz_model_by_enz_usage">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">change_enz_model_by_enz_usage</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">enz_ratio</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">json_model_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_flux_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">EC_max_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_kcat_mw</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">need_change_reaction_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">changed_reaction_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">f</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ptot</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sigma</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lowerbound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">upperbound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">json_output_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.change_enz_model_by_enz_usage" title="Permalink to this definition"></a></dt>
<dd><p>Get new enzyme model using enzyme usage to calibration</p>
<section id="id3">
<h2>Arguments<a class="headerlink" href="#id3" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>enz_ratio: enzyme ratio which needed change.</p></li>
<li><p>json_model_path: The file storing json model.</p></li>
<li><p>reaction_flux_file: reaction-flux file.</p></li>
<li><p>reaction_kcat_mw_file: reaction kcat/mw file.</p></li>
<li><p>reaction_enz_usage_file: enzyme usage of each reaction.</p></li>
<li><p>kcat_database_combined_file: combined kcat file got from autoPACMEN.</p></li>
<li><p>model_file: cobra model.</p></li>
<li><p>f: The enzyme mass fraction</p></li>
<li><p>ptot: The total protein fraction in cell.</p></li>
<li><p>sigma: The approximated average saturation of enzyme.</p></li>
<li><p>lowerbound:  Lowerbound  of enzyme concentration constraint.</p></li>
<li><p>upperbound:  Upperbound  of enzyme concentration constraint.</p></li>
<li><p>json_output_file: json file store json model</p></li>
<li><p>reaction_mw_outfile: changed file stored reaction kcat/mw.</p></li>
</ul>
<dl class="field-list simple">
<dt class="field-odd">return<span class="colon">:</span></dt>
<dd class="field-odd"><p>new enzyme model</p>
</dd>
</dl>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.change_reaction_kcat_by_database">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">change_reaction_kcat_by_database</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">json_model_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">select_reactionlist</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">EC_max_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_kcat_mw_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_kapp_change_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.change_reaction_kcat_by_database" title="Permalink to this definition"></a></dt>
<dd><p>Use the kcat in database to change reaction kcat in model</p>
<section id="id4">
<h2>Arguments<a class="headerlink" href="#id4" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>json_model_path: The file storing json model.</p></li>
<li><p>select_reactionlist: reaction list need to change.</p></li>
<li><p>kcat_database_combined_file: combined kcat file got from autoPACMEN.</p></li>
<li><p>reaction_kcat_mw_file: reaction kcat/mw file.</p></li>
<li><p>reaction_kapp_change_file: changed file stored reaction kcat/mw.</p></li>
</ul>
<dl class="field-list simple">
<dt class="field-odd">return<span class="colon">:</span></dt>
<dd class="field-odd"><p>a dataframe stored new reaction kcat/mw .</p>
</dd>
</dl>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.combine_reactions_simles_sequence">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">combine_reactions_simles_sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">metdf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">smilesdf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">prodf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">comdf_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.combine_reactions_simles_sequence" title="Permalink to this definition"></a></dt>
<dd><p>This function is used to combine the reaction–substrate–gene–protein_sequnce–mass.</p>
<p>Arguments:
* metdf: metabolites_reactions_gpr
* similesdf: inchkeydf
* prodf: prodf</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.convert_bigg_met_to_inchikey">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">convert_bigg_met_to_inchikey</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">metlist</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inchikey_list_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.convert_bigg_met_to_inchikey" title="Permalink to this definition"></a></dt>
<dd><p>This function is used to convert the bigg id of metabolites in the model to inchikey.</p>
<section id="id5">
<h2>Arguments<a class="headerlink" href="#id5" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>metlist: list ~ List of metabolite IDs.</p></li>
<li><p>inchikey_list_file: str ~ File path to save the resulting DataFrame as a CSV file.</p></li>
</ul>
</section>
<section id="returns">
<h2>Returns<a class="headerlink" href="#returns" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>inchkeydf: pandas.DataFrame ~ DataFrame containing metabolite names and InChIKeys.</p></li>
</ul>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.convert_inchikey_to_smiles">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">convert_inchikey_to_smiles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">inchkeydf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inchikey_list_smilesfile</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.convert_inchikey_to_smiles" title="Permalink to this definition"></a></dt>
<dd><p>This function is used to convert the InChIKey of metabolites in the model to SMILES.</p>
<section id="id6">
<h2>Arguments<a class="headerlink" href="#id6" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>inchkeydf: pandas.DataFrame ~ DataFrame object containing the InChIKey of metabolites.</p></li>
<li><p>inchikey_list_smilesfile: str ~ File path to save the resulting DataFrame as a CSV file.</p></li>
</ul>
</section>
<section id="id7">
<h2>Returns<a class="headerlink" href="#id7" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>inchkeydf: pandas.DataFrame ~ DataFrame with added ‘Canonical_SMILES’ column containing SMILES.</p></li>
</ul>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.convert_to_irreversible">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">convert_to_irreversible</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.convert_to_irreversible" title="Permalink to this definition"></a></dt>
<dd><p>Split reversible reactions into two irreversible reactions</p>
<p>These two reactions will proceed in opposite directions. This
guarentees that all reactions in the model will only allow
positive flux values, which is useful for some modeling problems.</p>
<section id="id8">
<h2>Arguments<a class="headerlink" href="#id8" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>model: cobra.Model ~ A Model object which will be modified in place.</p></li>
</ul>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.create_file">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">create_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">store_path</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.create_file" title="Permalink to this definition"></a></dt>
<dd><p>Create a directory at the specified path.</p>
<dl class="simple">
<dt>Args:</dt><dd><p>store_path (str): The path of the directory to create.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.create_trace">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">create_trace</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">color</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">symbol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xaxis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">yaxis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.create_trace" title="Permalink to this definition"></a></dt>
<dd><p>Create a scatter trace for a plot.</p>
<p>Arguments:
* x: list or array-like - x-axis data.
* y: list or array-like - y-axis data.
* name: str - Name of the trace.
* color: str - Color of the markers and line.
* size: int, optional - Size of the markers (default: 10).
* symbol: str, optional - Symbol for the markers (default: None).
* width: int or None, optional - Width of the line (default: None).
* xaxis: str or None, optional - Name of the x-axis (default: None).
* yaxis: str or None, optional - Name of the y-axis (default: None).</p>
<p>Returns:
* trace: go.Scatter - Scatter trace object.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.draw_3d_rbas">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">draw_3d_rbas</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">z_data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">substrate_bound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">o2_bound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">obj_bound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">number</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">PhPP_output_fig_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.draw_3d_rbas" title="Permalink to this definition"></a></dt>
<dd><p>Generate a 3D plot of reaction bounds analysis (RBA) results.</p>
<p>Arguments:
* z_data: pd.DataFrame - Dataframe containing the z-axis values for the plot.
* substrate_bound: float - Upper bound of the substrate.
* o2_bound: float - Upper bound of O2 uptake rates.
* obj_bound: float - Upper bound of growth rates.
* number: int - Number of data points.
* PhPP_output_fig_file: str - File path to save the generated figure.</p>
<p>Returns:
* fig: go.Figure - Generated 3D plot figure.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.draw_cdf_fig">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">draw_cdf_fig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data_cdf_data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y_index</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nticks</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.draw_cdf_fig" title="Permalink to this definition"></a></dt>
<dd><p>Generate a cumulative distribution function (CDF) plot.</p>
<p>Arguments:
* data_cdf_data: list - Data for the CDF plot.
* output_file: str - File path to save the generated figure.
* x_name: str - Name of the x-axis.
* y_name: str - Name of the y-axis.
* y_index: list - Index values for the y-axis.
* nticks: int - Number of ticks for the x-axis.</p>
<p>Returns:
* fig: go.Figure - Generated CDF plot figure.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.draw_overfolw_fig">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">draw_overfolw_fig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">GEMyield_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ecGEMyield_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">column_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y_axis_loc_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">color_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">substrate_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">substrate_bound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">obj_bound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">secrate_bound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">trade_off_biomass_yield_figfile</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.draw_overfolw_fig" title="Permalink to this definition"></a></dt>
<dd><p>Create a figure showing trade-off between biomass yield and substrate/secretion rates.</p>
<p>Arguments:
* GEMyield_list: pandas DataFrame - Data for GEM yields.
* ecGEMyield_list: pandas DataFrame - Data for ecGEM yields.
* column_list: list - List of column names.
* y_axis_loc_list: list - List of y-axis locations (‘left’ or ‘right’) for each column.
* color_list: list - List of colors for the traces.
* substrate_bound: float - Upper limit for substrate uptake rate.
* obj_bound: float - Upper limit for growth rate.
* secrate_bound: float - Upper limit for secretion rate.
* trade_off_biomass_yield_figfile: str - Output file name for the figure.</p>
<p>Returns:
* fig: go.Figure - Plotly figure object.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.draw_trade_off">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">draw_trade_off</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">efficiency_pfba</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">trade_off_enzyme_efficiency_figfile</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.draw_trade_off" title="Permalink to this definition"></a></dt>
<dd><p>Draw a trade-off plot showing the relationship between biomass yield and enzyme efficiency.</p>
<p>Arguments:
* efficiency_pfba: pd.DataFrame - DataFrame containing efficiency data.
* trade_off_enzyme_efficiency_figfile: str - File path to save the trade-off plot.</p>
<p>Returns:
* fig: go.Figure - Plotly Figure object.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.drawphpp">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">drawphpp</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">GEM_glc_o2_df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ecGEM_glc_o2_df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">substrate_bound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">o2_bound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">obj_bound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">number</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">PhPP_output_fig_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.drawphpp" title="Permalink to this definition"></a></dt>
<dd><p>Generate a 3D surface plot of PhPP results for GEM and ecGEM.</p>
<p>Arguments:
* GEM_glc_o2_df: pd.DataFrame - DataFrame containing GEM growth rates for different glucose and oxygen uptake rates.
* ecGEM_glc_o2_df: pd.DataFrame - DataFrame containing ecGEM growth rates for different glucose and oxygen uptake rates.
* substrate_bound: float - Upper bound for the substrate uptake rates.
* o2_bound: float - Upper bound for the oxygen uptake rates.
* obj_bound: float - Upper bound for the growth rates.
* number: int - Number of intervals to divide the bounds.
* PhPP_output_fig_file: str - File path to save the generated figure.</p>
<p>Returns:
* fig: go.Figure - Generated 3D surface plot figure.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.extract_metabolite_id">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">extract_metabolite_id</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">met_id</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.extract_metabolite_id" title="Permalink to this definition"></a></dt>
<dd><p>Extracts the metabolite ID without compartment information.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.generate_DLKCAT_input">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">generate_DLKCAT_input</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">metdf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metdf_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metdf_outfile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">DLinputdf_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.generate_DLKCAT_input" title="Permalink to this definition"></a></dt>
<dd><p>This function is used to generate the DLKCAT input file.</p>
<p>Arguments:
* metdf: pandas.DataFrame - DataFrame containing reaction, substrate, gene, protein sequence, and mass information.
* metdf_name: pandas.DataFrame - DataFrame containing metabolite names.
* metdf_outfile: str - File path to save the modified metdf DataFrame as a CSV file.
* DLinputdf_file: str - File path to save the DLKCAT input DataFrame as a CSV file.</p>
<p>Returns:
* DLinputdf: pandas.DataFrame - DataFrame containing DLKCAT input data.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.generate_random_colors">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">generate_random_colors</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">num_colors</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.generate_random_colors" title="Permalink to this definition"></a></dt>
<dd><p>Generate a list of random colors in hexadecimal format.</p>
<p>Arguments:
* num_colors: int - Number of colors to generate.</p>
<p>Returns:
* colors: list - List of random colors in hexadecimal format.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_PhPP_data">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_PhPP_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_type</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">obj</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">substrate_bound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">o2_bound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">number</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">outputfile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_id</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">z_id</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_PhPP_data" title="Permalink to this definition"></a></dt>
<dd><p>Perform PhPP analysis on a model using different substrate and oxygen bounds.</p>
<p>Arguments:
* model_file: str - File path to the model file (JSON or SBML format).
* model_type: str - Type of the model (‘GEM’ for Genome-Scale Model, other types for enzyme-constraint models).
* obj: str - ID of the objective reaction in the model.
* substrate_bound: float - Upper bound for the substrate constraint.
* o2_bound: float - Upper bound for the oxygen constraint.
* number: int - Number of intervals to divide the bounds.
* outputfile: str - File path to save the resulting DataFrame as a CSV file.
* x_id: str - ID of the first reaction to constrain.
* z_id: str - ID of the second reaction to constrain.</p>
<p>Returns:
* df1: pd.DataFrame - DataFrame containing the results of PhPP analysis.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_currency_metabolites">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_currency_metabolites</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_currency_metabolites" title="Permalink to this definition"></a></dt>
<dd><p>Returns a set of currency metabolites.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_enz_foldchange">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_enz_foldchange</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ecModel_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">obj</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">substrate</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">substrate_con</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">biomass_id</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">biomass_min</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">biomass_max</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">FC_threshold</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fluxes_outfile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">enzcost_diff_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_enz_foldchange" title="Permalink to this definition"></a></dt>
<dd><p>Calculate the enzyme fold change between the minimum and maximum biomass conditions.</p>
<p>Arguments:
* ecModel_file: str - File path to the enzyme-constrained model.
* obj: str - ID of the objective reaction in the model.
* substrate: str - ID of the substrate reaction.
* substrate_con: float - Substrate concentration.
* biomass_id: str - ID of the biomass reaction.
* biomass_min: float - Minimum biomass value.
* biomass_max: float - Maximum biomass value.
* fluxes_outfile: str - File path to save the fluxes data.
* enzcost_diff_file: str - File path to save the enzyme cost difference.</p>
<p>Returns:
* enzcost_diff: pd.DataFrame - DataFrame containing enzyme cost difference data.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_enz_model_use_enz_usage_by_eckcat">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_enz_model_use_enz_usage_by_eckcat</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">enz_ratio</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">json_model_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_flux_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">EC_max_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_kcat_mw_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">f</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ptot</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sigma</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lowerbound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">upperbound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">json_output_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_mw_outfile</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_enz_model_use_enz_usage_by_eckcat" title="Permalink to this definition"></a></dt>
<dd><p>Get new enzyme model using enzyme usage to calibration</p>
<section id="id9">
<h2>Arguments<a class="headerlink" href="#id9" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>enz_ratio: enzyme ratio which needed change.</p></li>
<li><p>json_model_path: The file storing json model.</p></li>
<li><p>reaction_flux_file: reaction-flux file.</p></li>
<li><p>reaction_kcat_mw_file: reaction kcat/mw file.</p></li>
<li><p>reaction_enz_usage_file: enzyme usage of each reaction.</p></li>
<li><p>kcat_database_combined_file: combined kcat file got from autoPACMEN.</p></li>
<li><p>model_file: cobra model.</p></li>
<li><p>f: The enzyme mass fraction</p></li>
<li><p>ptot: The total protein fraction in cell.</p></li>
<li><p>sigma: The approximated average saturation of enzyme.</p></li>
<li><p>lowerbound:  Lowerbound  of enzyme concentration constraint.</p></li>
<li><p>upperbound:  Upperbound  of enzyme concentration constraint.</p></li>
<li><p>json_output_file: json file store json model</p></li>
<li><p>reaction_mw_outfile: changed file stored reaction kcat/mw.</p></li>
</ul>
<dl class="field-list simple">
<dt class="field-odd">return<span class="colon">:</span></dt>
<dd class="field-odd"><p>new enzyme model</p>
</dd>
</dl>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_enzyme_constraint_model">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_enzyme_constraint_model</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">json_model_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_enzyme_constraint_model" title="Permalink to this definition"></a></dt>
<dd><p>using enzyme concentration constraint
json model to create a COBRApy model.</p>
<section id="id10">
<h2>Arguments<a class="headerlink" href="#id10" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>json_model_file: json Model file.</p></li>
</ul>
<dl class="field-list simple">
<dt class="field-odd">return<span class="colon">:</span></dt>
<dd class="field-odd"><p>Construct an enzyme-constrained model.</p>
</dd>
</dl>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_enzyme_constraint_model_percent">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_enzyme_constraint_model_percent</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">json_model_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">percent</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_enzyme_constraint_model_percent" title="Permalink to this definition"></a></dt>
<dd><p>using enzyme concentration constraint
json model to create a COBRApy model.</p>
<section id="id11">
<h2>Arguments<a class="headerlink" href="#id11" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>json_model_file: json Model file.</p></li>
</ul>
<dl class="field-list simple">
<dt class="field-odd">return<span class="colon">:</span></dt>
<dd class="field-odd"><p>Construct an enzyme-constrained model.</p>
</dd>
</dl>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_fluxes_detail_in_model">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_fluxes_detail_in_model</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_pfba_solution</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fluxes_outfile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">json_model_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_fluxes_detail_in_model" title="Permalink to this definition"></a></dt>
<dd><p>Get the detailed information of each reaction.</p>
<p>Arguments:
* model: cobra.Model - The metabolic model.
* model_pfba_solution: pandas.Series - The pFBA solution containing reaction fluxes.
* fluxes_outfile: str - Path to the output file for reaction fluxes.
* json_model_file: str - Path to the JSON model file.</p>
<p>Returns:
* model_pfba_solution_detail: pandas.DataFrame - Detailed information of each reaction.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_gene_subunitDescription">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_gene_subunitDescription</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sub_description_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_gene_subunitDescription" title="Permalink to this definition"></a></dt>
<dd><p>Get gene subunit descriptions from UniProt for a given model.</p>
<p>Arguments:
* sub_description_path: str - File path to save the resulting DataFrame as a CSV file.
* model: cobra.Model - A Model object.</p>
<p>Returns:
None</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_genes_and_gpr">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_genes_and_gpr</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_outfile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gpr_outfile</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_genes_and_gpr" title="Permalink to this definition"></a></dt>
<dd><p>Retrieve genes and gene_reaction_rule from a Genome-Scale Metabolic Network Model.</p>
<dl class="simple">
<dt>Args:</dt><dd><p>model (cobra.Model): A genome-scale metabolic network model.
gene_outfile (str): Path to the output file to save the genes.
gpr_outfile (str): Path to the output file to save the gene_reaction_rule.</p>
</dd>
<dt>Returns:</dt><dd><p>list: A list containing two pandas DataFrames - genes and gpr.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_met_bigg_id">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_met_bigg_id</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_met_bigg_id" title="Permalink to this definition"></a></dt>
<dd><p>This function is used to get the bigg id of metabolites in the model.</p>
<section id="id12">
<h2>Arguments<a class="headerlink" href="#id12" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>model: cobra.Model ~ A Model object which will be modified in place.</p></li>
</ul>
</section>
<section id="id13">
<h2>Returns<a class="headerlink" href="#id13" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>metdf_unique: pandas.DataFrame ~ A DataFrame containing unique metabolite IDs and names.</p></li>
</ul>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_min_enzyme_cost">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_min_enzyme_cost</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dict_coeff</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_min_enzyme_cost" title="Permalink to this definition"></a></dt>
<dd><p>Get model flux using Minimum enzyme cost algorithm</p>
<section id="id14">
<h2>Arguments<a class="headerlink" href="#id14" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>model: cobra model.</p></li>
<li><p>dict_coeff: {reaction ID: coeffient}.</p></li>
</ul>
<dl class="field-list simple">
<dt class="field-odd">return<span class="colon">:</span></dt>
<dd class="field-odd"><p>cobra solution.</p>
</dd>
</dl>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_model_protein_sequence_and_mass">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_model_protein_sequence_and_mass</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">subbnumdf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">prodf_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_model_protein_sequence_and_mass" title="Permalink to this definition"></a></dt>
<dd><p>This function is used to get the protein sequence and mass of the model.</p>
<section id="id15">
<h2>Arguments<a class="headerlink" href="#id15" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>model: cobra.Model ~ A Model object which will be modified in place.</p></li>
<li><p>subbnumdf: pandas.DataFrame ~ DataFrame containing enzyme subunit information.</p></li>
<li><p>prodf_file: str ~ File path to save the resulting DataFrame as a CSV file.</p></li>
</ul>
</section>
<section id="id16">
<h2>Returns<a class="headerlink" href="#id16" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>prodf: pandas.DataFrame ~ DataFrame with protein information.</p></li>
</ul>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_model_substrate_obj">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_model_substrate_obj</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">use_model</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_model_substrate_obj" title="Permalink to this definition"></a></dt>
<dd><p>change model substrate for single carbon source</p>
<section id="id17">
<h2>Arguments<a class="headerlink" href="#id17" title="Permalink to this heading"></a></h2>
<p>use_model: cobra.Model ~ A Model object which will be modified in place.</p>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_protein_mass_mapping_from_local">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_protein_mass_mapping_from_local</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sbml_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">project_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">project_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">uniprot_data_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ECMpy_function.get_protein_mass_mapping_from_local" title="Permalink to this definition"></a></dt>
<dd><p>Returns a JSON with a mapping of protein IDs as keys, and as values the protein mass in kDa.</p>
<p>The protein masses are calculated using the amino acid sequence from UniProt (retrieved using
UniProt’s REST API).</p>
<section id="id18">
<h2>Arguments<a class="headerlink" href="#id18" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>model: cobra.Model ~ The model in the cobrapy format</p></li>
<li><p>project_folder: str ~ The folder in which the JSON shall be created</p></li>
<li><p>project_name: str ~ The beginning of the JSON’s file name</p></li>
<li><p>uniprot_data_file: str ~ The gene information obtained from uniprot</p></li>
</ul>
</section>
<section id="output">
<h2>Output<a class="headerlink" href="#output" title="Permalink to this heading"></a></h2>
<p>A JSON file with the path project_folder+project_name+’_protein_id_mass_mapping.json’
and the following structure:
&lt;pre&gt;
{</p>
<blockquote>
<div><p>“$PROTEIN_ID”: $PROTEIN_MASS_IN_KDA,
(…),</p>
</div></blockquote>
<p>}
&lt;/pre&gt;</p>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_reaction_kcat_mw">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_reaction_kcat_mw</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">project_folder</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">project_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">type_of_default_kcat_selection</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">enzyme_unit_number_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">json_output_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_reaction_kcat_mw" title="Permalink to this definition"></a></dt>
<dd><p>Adds proteomic constraints according to sMOMENT to the given stoichiometric model and stores it as SBML.</p>
<section id="id19">
<h2>Arguments:<a class="headerlink" href="#id19" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>model: cobra.Model ~ A cobra Model representation of the metabolic network. This model will
be changed using cobrapy functions in order to add the proteomic constraints.</p></li>
<li><p>project_folder: str ~ The folder in which the spreadsheets and JSONs with the model’s supplemental
data can be found.</p></li>
<li><p>project_name: str ~ The sMOMENTed model creation’s name, which will be added at the beginning
of the created SBML’s name.</p></li>
<li><p>type_of_default_kcat_selection: str ~ The type of default kcat selection to use. Available options
are “median”, “mean”, “max”, “random”, or “Null”.</p></li>
<li><p>enzyme_unit_number_file: str ~ The file path of the enzyme unit number data.</p></li>
<li><p>json_output_file: str ~ The file path to save the output as a JSON file.</p></li>
</ul>
</section>
<section id="id20">
<h2>Output:<a class="headerlink" href="#id20" title="Permalink to this heading"></a></h2>
<p>None.</p>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_reaction_kcatmw_onestop">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_reaction_kcatmw_onestop</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">work_folder</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">kcat_method</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bigg_metabolites_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">brenda_textfile_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">uniprot_data_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">organism</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">kcat_gap_fill</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_gap_fill</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_reaction_kcatmw_onestop" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_reaction_kcatmw_onestop_by_AutoPACMEN">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_reaction_kcatmw_onestop_by_AutoPACMEN</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">autopacmen_folder</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sbml_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bigg_metabolites_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">brenda_textfile_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">project_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">uniprot_data_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">organism</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_kcat_database_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">kcat_gap_fill</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_gap_fill</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_reaction_kcatmw_onestop_by_AutoPACMEN" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_reaction_kcatmw_onestop_by_DLKcat">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_reaction_kcatmw_onestop_by_DLKcat</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dlkcat_folder</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sbml_path</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_reaction_kcatmw_onestop_by_DLKcat" title="Permalink to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_reaction_mw">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_reaction_mw</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sbml_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">project_folder</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">project_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">json_output_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">enzyme_unit_number_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_reaction_mw" title="Permalink to this definition"></a></dt>
<dd><p>Adds proteomic constraints according to sMOMENT to the given stoichiometric model and stores it as SBML.</p>
<p>Arguments:
* sbml_path: str - Path to the SBML or JSON file representing the model.
* project_folder: str - The folder in which the spreadsheets and JSONs with the model’s supplemental</p>
<blockquote>
<div><p>data can be found.</p>
</div></blockquote>
<ul class="simple">
<li><p>project_name: str - The sMOMENTed model creation’s name, which will be added at the beginning
of the created SBML’s name.</p></li>
<li><p>json_output_file: str - Path to the output JSON file to store the reaction molecular weights.</p></li>
<li><p>enzyme_unit_number_file: str - Path to the enzyme unit number file.</p></li>
</ul>
<section id="id21">
<h2>Output:<a class="headerlink" href="#id21" title="Permalink to this heading"></a></h2>
<p>Writes the reaction molecular weights to a JSON file.</p>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_subunit_number">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_subunit_number</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sub_description_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_subnum_path</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_subunit_number" title="Permalink to this definition"></a></dt>
<dd><p>Get the number of subunits for each gene from the subunit description DataFrame.</p>
<p>Arguments:
* sub_description_path: str - File path to the subunit description DataFrame (CSV file).
* gene_subnum_path: str - File path to save the resulting DataFrame with subunit numbers as a CSV file.</p>
<p>Returns:
* df: pd.DataFrame - DataFrame with subunit numbers for each gene.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.get_yield_cost_efficiency">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">get_yield_cost_efficiency</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">enz_model</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">glc_concentration_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_substrate</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">obj</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_kcat_MW</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">efficiency_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.get_yield_cost_efficiency" title="Permalink to this definition"></a></dt>
<dd><p>Calculate yield, cost, and efficiency for different glucose concentrations.</p>
<p>Arguments:
* enz_model: cobra.Model - Enzyme-constrained metabolic model.
* glc_concentration_list: list - List of glucose concentrations to evaluate.
* use_substrate: str - ID of the substrate reaction in the model.
* obj: str - ID of the objective reaction in the model.
* reaction_kcat_MW: dict - Dictionary mapping reaction IDs to kcat/MW values.
* efficiency_file: str - File path to save the efficiency data.</p>
<p>Returns:
* yield_cost_efficiency_df: pd.DataFrame - DataFrame containing yield, cost, and efficiency values.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.isoenzyme_split">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">isoenzyme_split</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.isoenzyme_split" title="Permalink to this definition"></a></dt>
<dd><p>Split isoenzyme reaction to mutiple reaction</p>
<section id="id22">
<h2>Arguments<a class="headerlink" href="#id22" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>model: cobra.Model.</p></li>
</ul>
<dl class="field-list simple">
<dt class="field-odd">return<span class="colon">:</span></dt>
<dd class="field-odd"><p>new cobra.Model.</p>
</dd>
</dl>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.json_load">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">json_load</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">Any</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Any</span><span class="p"><span class="pre">]</span></span></span></span><a class="headerlink" href="#ECMpy_function.json_load" title="Permalink to this definition"></a></dt>
<dd><p>Loads the given JSON file and returns it as dictionary.</p>
<section id="id23">
<h2>Arguments<a class="headerlink" href="#id23" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>path: str ~ The path of the JSON file</p></li>
</ul>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.json_write">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">json_write</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dictionary</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.json_write" title="Permalink to this definition"></a></dt>
<dd><p>Writes a JSON file at the given path with the given dictionary as content.</p>
<section id="id24">
<h2>Arguments<a class="headerlink" href="#id24" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>path:   The path of the JSON file that shall be written</p></li>
<li><p>dictionary: The dictionary which shalll be the content of
the created JSON file</p></li>
</ul>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.parse_sabio_rk_for_eclist">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">parse_sabio_rk_for_eclist</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ec_numbers_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">json_output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bigg_id_name_mapping_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ECMpy_function.parse_sabio_rk_for_eclist" title="Permalink to this definition"></a></dt>
<dd><p>Retrieves kcats from SABIO-RK for the given model and stores it in a JSON for the given model in the given path.</p>
<section id="algorithm">
<h2>Algorithm<a class="headerlink" href="#algorithm" title="Permalink to this heading"></a></h2>
<p>Using the SABIO-RK REST API (as of 2019/30/04, it is explained under
<a class="reference external" href="http://sabiork.h-its.org/layouts/content/docuRESTfulWeb/RESTWebserviceIntro.gsp">http://sabiork.h-its.org/layouts/content/docuRESTfulWeb/RESTWebserviceIntro.gsp</a>),</p>
</section>
<section id="id25">
<h2>Arguments<a class="headerlink" href="#id25" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>eclist: List[str] ~ eclist.</p></li>
<li><p>json_output_path: str ~ The path of the JSON that shall be created</p></li>
</ul>
</section>
<section id="id26">
<h2>Output<a class="headerlink" href="#id26" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>A JSON in the given project folder with the following structure:</p></li>
</ul>
<dl>
<dt>&lt;pre&gt;</dt><dd><dl>
<dt>{</dt><dd><dl>
<dt>“$EC_NUMBER_OR_KEGG_REACTION_ID”: {</dt><dd><dl>
<dt>“$SUBSTRATE_WITH_BIGG_ID_1”: {</dt><dd><dl class="simple">
<dt>“$ORGANISM_1”: [</dt><dd><p>$kcat_1,
(…)
$kcat_n,</p>
</dd>
</dl>
<p>]</p>
</dd>
</dl>
<p>},
(…),
“REST”: {</p>
<blockquote>
<div><dl class="simple">
<dt>“$ORGANISM_1”: [</dt><dd><p>$kcat_1,
(…)
$kcat_n,</p>
</dd>
</dl>
<p>]</p>
</div></blockquote>
<p>}</p>
</dd>
</dl>
<p>}
(…),</p>
</dd>
</dl>
<p>}</p>
</dd>
</dl>
<p>&lt;/pre&gt;
‘REST’ stands for a substrate without found BIGG ID.</p>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.preprocess_metabolites">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">preprocess_metabolites</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">currency_metabolites</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.preprocess_metabolites" title="Permalink to this definition"></a></dt>
<dd><p>Preprocesses the metabolites and creates a DataFrame.</p>
<section id="id27">
<h2>Arguments<a class="headerlink" href="#id27" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>model: cobra.Model ~ A Model object to extract metabolite information from.</p></li>
<li><p>currency_metabolites: set ~ A set of currency metabolite IDs.</p></li>
</ul>
</section>
<section id="id28">
<h2>Returns<a class="headerlink" href="#id28" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>metdf: pandas.DataFrame ~ DataFrame with preprocessed metabolite information.</p></li>
</ul>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.run_FSEOF">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">run_FSEOF</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">substrate</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">substrate_con</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">biomass_id</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">obj</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">FSEOF_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.run_FSEOF" title="Permalink to this definition"></a></dt>
<dd><p>Perform Flux Scanning-based Enzyme-Feasible Flux Optimization (FSEOF) analysis.</p>
<p>Arguments:
* model: cobra.Model - COBRA model object.
* substrate: str - ID of the substrate reaction.
* substrate_con: float - Substrate concentration.
* biomass_id: str - ID of the biomass reaction.
* obj: str - ID of the objective reaction.</p>
<p>Returns:
* FSEOFdf_f: pd.DataFrame - DataFrame containing FSEOF analysis results.
* FESEOF_gene: pd.DataFrame - DataFrame containing genes and their regulations.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.split_substrate_to_match_gene">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">split_substrate_to_match_gene</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metabolites_reactions_gpr_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.split_substrate_to_match_gene" title="Permalink to this definition"></a></dt>
<dd><p>Splits the substrate of reactions to match the gene of reactions and saves the results to a file.</p>
<section id="id29">
<h2>Arguments<a class="headerlink" href="#id29" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>model: cobra.Model ~ A Model object which will be modified in place.</p></li>
<li><p>metabolites_reactions_gpr_file: str ~ File path to save the resulting DataFrame as a CSV file.</p></li>
</ul>
</section>
<section id="id30">
<h2>Returns<a class="headerlink" href="#id30" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>metdf: pandas.DataFrame ~ DataFrame with metabolite information.</p></li>
</ul>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.standardize_folder">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">standardize_folder</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ECMpy_function.standardize_folder" title="Permalink to this definition"></a></dt>
<dd><p>Returns for the given folder path is returned in a more standardized way.</p>
<p>I.e., folder paths with potential are replaced with /. In addition, if
a path does not end with / will get an added /.</p>
<section id="argument">
<h2>Argument<a class="headerlink" href="#argument" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>folder: str ~ The folder path that shall be standardized.</p></li>
</ul>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ECMpy_function.trans_model2enz_json_model_split_isoenzyme">
<span class="sig-prename descclassname"><span class="pre">ECMpy_function.</span></span><span class="sig-name descname"><span class="pre">trans_model2enz_json_model_split_isoenzyme</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reaction_kcat_mw_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">f</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ptot</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sigma</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lowerbound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">upperbound</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">json_output_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ECMpy_function.trans_model2enz_json_model_split_isoenzyme" title="Permalink to this definition"></a></dt>
<dd><p>Tansform cobra model to json mode with  
enzyme concentration constraintat.</p>
<section id="id31">
<h2>Arguments<a class="headerlink" href="#id31" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>model_file:   The path of sbml model</p></li>
<li><p>reaction_kcat_mw_file: The path of storing kcat/MW value of the enzyme catalyzing each</p></li>
</ul>
<blockquote>
<div><p>reaction in the GEM model</p>
</div></blockquote>
<ul class="simple">
<li><p>f: The enzyme mass fraction</p></li>
<li><p>ptot: The total protein fraction in cell.</p></li>
<li><p>sigma: The approximated average saturation of enzyme.</p></li>
<li><p>lowerbound:  Lowerbound  of enzyme concentration constraint.</p></li>
<li><p>upperbound:  Upperbound  of enzyme concentration constraint.</p></li>
</ul>
</section>
</dd></dl>

</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="AutoPACMEN_function.html" class="btn btn-neutral float-left" title="AutoPACMEN_function" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="get_ecGEM_onestop.html" class="btn btn-neutral float-right" title="get_ecGEM_onestop" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Biodesign Center.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>