Search.setIndex({"docnames": ["AutoPACMEN_function", "ECMpy_function", "get_ecGEM_onestop", "index", "model", "prediction_for_input"], "filenames": ["AutoPACMEN_function.rst", "ECMpy_function.rst", "get_ecGEM_onestop.rst", "index.rst", "model.rst", "prediction_for_input.rst"], "titles": ["AutoPACMEN_function", "ECMpy_function", "get_ecGEM_onestop", "Documentation for ecmpy", "model", "prediction_for_input"], "terms": {"create_combined_kcat_databas": [0, 3], "sabio_rk_kcat_database_path": 0, "str": [0, 1], "brenda_kcat_database_path": 0, "output_path": 0, "none": [0, 1], "creat": [0, 1, 5], "combin": [0, 1], "json": [0, 1], "given": [0, 1], "sabio": [0, 1], "k": 0, "brenda": 0, "kcat": [0, 1], "databas": [0, 1], "non": 0, "wildcard": 0, "entri": 0, "onli": [0, 1], "The": [0, 1, 4], "path": [0, 1], "rk": [0, 1], "outputh": 0, "filenam": [0, 4, 5], "genreat": 0, "A": [0, 1, 4], "follow": [0, 1], "format": [0, 1], "pre": [0, 1], "ec_numb": 0, "bigg_ids_of_substr": 0, "organ": [0, 1], "float": [0, 1], "sourc": [0, 1], "sabio_rk": 0, "fals": 0, "create_fil": [0, 1, 3], "store_path": [0, 1], "ensure_folder_exist": [0, 3], "folder": [0, 1], "check": 0, "exist": 0, "If": [0, 1], "i": [0, 1, 4, 5], "whose": 0, "shall": [0, 1], "enforc": 0, "get_ec_number_kcats_wildcard_search": [0, 3], "list": [0, 1, 5], "bigg_id_name_mapping_path": [0, 1], "batch_siz": 0, "int": [0, 1], "5": 0, "dict": [0, 1], "ani": [0, 1], "return": 0, "ec": 0, "number": [0, 1], "depend": 0, "us": [0, 1, 5], "an": [0, 1], "increment": 0, "level": 0, "until": 0, "kcart": 0, "were": 0, "found": [0, 1], "all": [0, 1, 4], "full": 0, "bigg": [0, 1], "id": [0, 1, 5], "name": [0, 1, 5], "map": [0, 1], "how": 0, "mani": 0, "look": 0, "up": 0, "parallel": 0, "do": 0, "set": [0, 1], "thi": [0, 1, 4], "too": 0, "high": 0, "dictionari": [0, 1, 5], "content": [0, 1], "ec_number_or_kegg_reaction_id": [0, 1], "substrate_with_bigg_id_1": [0, 1], "organism_1": [0, 1], "kcat_1": [0, 1], "kcat_n": [0, 1], "rest": [0, 1], "stand": [0, 1], "substrat": [0, 1], "without": [0, 1], "get_entrez_id_from_organism_full_nam": [0, 3], "organism_full_nam": 0, "get": [0, 1], "": [0, 1], "entrez": 0, "numer": 0, "identifi": 0, "neccessari": 0, "blast": 0, "ncbi": 0, "taxonomi": [0, 1], "search": [0, 3], "function": [0, 1, 4], "biopython": 0, "compat": 0, "txid": 0, "orgn": 0, "organism_kegg_id": 0, "e": [0, 1, 5], "g": [0, 5], "xanthomona": 0, "campestri": 0, "pv": 0, "campesri": 0, "b100": 0, "get_entrez_id_from_organism_full_name_batch": [0, 3], "retriev": [0, 1], "get_fil": [0, 3], "file": [0, 1], "string": 0, "which": [0, 1, 5], "get_id_associated_kcat": [0, 3], "searched_id": 0, "id_typ": 0, "data": [0, 1, 5], "kegg": 0, "call": [0, 4], "api": [0, 1], "must": 0, "either": 0, "whether": 0, "you": 0, "ar": [0, 1], "batch": 0, "satch_siz": 0, "five": 0, "onc": 0, "get_protein_mass_map": [0, 3], "model": [0, 1, 3, 5], "project_fold": [0, 1], "project_nam": [0, 1], "protein": [0, 1], "kei": [0, 1, 5], "valu": [0, 1, 5], "mass": [0, 1], "kda": [0, 1], "calcul": [0, 1], "amino": [0, 1], "acid": [0, 1], "sequenc": [0, 1, 5], "from": [0, 1, 5], "uniprot": [0, 1], "cobra": [0, 1], "cobrapi": [0, 1], "begin": [0, 1], "_protein_id_mass_map": [0, 1], "structur": [0, 1], "protein_id": [0, 1], "protein_mass_in_kda": [0, 1], "get_protein_mass_mapping_with_sbml": [0, 3], "sbml_path": [0, 1], "modul": [0, 3, 4], "sbml": [0, 1], "instead": [0, 4], "get_reactions_kcat_map": [0, 3], "kcat_database_path": 0, "sa_database_path": 0, "reaction_mw_path": 0, "protein_kcat_database_path": [0, 1], "type_of_kcat_select": 0, "reaction": [0, 1], "d": 0, "select": [0, 1], "affect": 0, "metabolit": [0, 1], "direct": [0, 1], "one": [0, 4], "each": [0, 1, 5], "forward": [0, 3, 4], "revers": [0, 1], "taxonom": 0, "nearest": 0, "prefer": 0, "te": 0, "sore": 0, "project": [0, 1], "alreadi": 0, "sa": 0, "mr": 0, "mao": 0, "custom": 0, "can": [0, 1], "mean": [0, 1], "median": [0, 1], "random": [0, 1], "refer": 0, "default": [0, 1], "_reactions_kcat_mapping_combin": 0, "reaction_nam": 0, "forward_kcat": 0, "reverse_kcat": 0, "get_taxonomy_from_organism_ncbi_id": [0, 3], "organism_ncbi_id": 0, "start": 0, "end": [0, 1], "highest": 0, "abov": 0, "ncbi_organism_id": 0, "get_taxonomy_from_organism_ncbi_id_batch": [0, 3], "where": 0, "is_fitting_ec_numb": [0, 3], "ec_number_on": 0, "ec_number_two": 0, "wildcard_level": 0, "bool": 0, "same": 0, "under": [0, 1], "first": [0, 1], "second": [0, 1], "json_load": [0, 1, 3], "load": [0, 1], "json_writ": [0, 1, 3], "write": [0, 1], "written": [0, 1], "shalll": [0, 1], "most_taxonomic_similar": [0, 3], "base_speci": 0, "taxonomy_dict": 0, "score": 0, "distanc": 0, "escherichia": 0, "coli": 0, "bacteria": 0, "pseudomona": 0, "homo": 0, "sapien": 0, "mammalia": 0, "animalia": 0, "would": 0, "0": 0, "1": 0, "2": 0, "speci": 0, "relat": 0, "made": 0, "sort": 0, "farthest": 0, "parse_bigg_metabolites_fil": [0, 3], "bigg_metabolites_file_path": 0, "json_output_fold": 0, "pars": 0, "text": 0, "As": 0, "29": 0, "04": [0, 1], "2019": [0, 1], "includ": 0, "http": [0, 1], "ucsd": 0, "edu": 0, "data_access": 0, "usual": 0, "ha": 0, "bigg_models_metabolit": 0, "txt": 0, "output_fold": 0, "store": [0, 1], "bigg_id_name_map": 0, "bigg_id": 0, "chemical_or_usual_nam": 0, "done": 0, "parse_brenda_json_for_model": [0, 3], "brenda_json_path": 0, "output_json_path": 0, "read": 0, "out": 0, "parse_brenda_textfil": [0, 3], "specif": 0, "newli": 0, "kcat_database_brenda": 0, "bigg_id_metabolit": 0, "kcat_list": 0, "brenda_textfile_path": [0, 1], "bigg_metabolites_json_fold": 0, "json_output_path": [0, 1], "json_output_path2": 0, "goe": 0, "through": 0, "textfil": 0, "convert": [0, 1], "machin": 0, "readabl": 0, "download": 0, "www": 0, "enzym": [0, 1], "org": [0, 1], "download_brenda_without_registr": 0, "php": 0, "complet": 0, "standard": [0, 1], "so": 0, "contain": [0, 1], "convolut": 0, "circumvent": 0, "mutat": 0, "exclud": 0, "have": 0, "parse_sabio_rk_for_eclist": [0, 1, 3], "ec_numbers_list": [0, 1], "30": [0, 1], "explain": [0, 1], "sabiork": [0, 1], "h": [0, 1], "its": [0, 1, 5], "layout": [0, 1], "docurestfulweb": [0, 1], "restwebserviceintro": [0, 1], "gsp": [0, 1], "eclist": [0, 1], "parse_sabio_rk_for_model": [0, 3], "parse_sabio_rk_for_model_with_sbml": [0, 3], "see": 0, "document": 0, "pickle_load": [0, 3], "pickl": 0, "pickle_writ": [0, 3], "pickled_object": 0, "object": [0, 1, 4, 5], "save": [0, 1], "sabio_rk_query_get_csv_lin": [0, 3], "query_dict": 0, "result": [0, 1], "queri": [0, 1], "csvreader": 0, "line": [0, 1], "form": 0, "id_nam": 0, "parametertyp": 0, "enzymetyp": 0, "wildtyp": 0, "sabio_rk_query_with_str": [0, 3], "differ": [0, 1], "no_result": 0, "esult": 0, "wa": 0, "csv": [0, 1], "sabio_rk_query_with_query_dict": [0, 3], "perform": [0, 1, 4], "them": [0, 4], "query_str": 0, "actual": 0, "term": 0, "request": 0, "could": 0, "like": [0, 1], "futher": 0, "process": 0, "python": 0, "librari": 0, "dictread": 0, "field": 0, "paramet": 0, "categori": 0, "possibl": 0, "standardize_fold": [0, 1, 3], "more": [0, 1], "wai": [0, 1], "potenti": [0, 1], "replac": [0, 1], "In": [0, 1], "addit": [0, 1], "doe": [0, 1], "ad": [0, 1], "dl_kcat_mw_calcul": [1, 3], "dlouputdf": 1, "metdf": 1, "make": 1, "kcat_mw": 1, "input": [1, 4], "ecgem": 1, "construct": 1, "panda": 1, "datafram": 1, "dlinputdf": 1, "dlkcat": 1, "gene": 1, "simil": 1, "prosequ": 1, "inform": 1, "determine_suitable_ecgem": [1, 3], "model_fil": 1, "bigg_met_fil": 1, "determin": 1, "provid": 1, "suitabl": 1, "constrain": 1, "tab": 1, "separ": 1, "bound": 1, "error": 1, "messag": 1, "genename_2_acc_from_uniprot": [1, 3], "outfil": 1, "amin": 1, "acc": 1, "adj_reaction_kcat_by_databas": [1, 3], "json_model_path": 1, "select_reactionlist": 1, "need_change_reaction_list": 1, "changed_reaction_list": 1, "ec_max_fil": 1, "reaction_kcat_mw": 1, "chang": 1, "need": [1, 4], "kcat_database_combined_fil": 1, "got": 1, "autopacmen": 1, "reaction_kcat_mw_fil": 1, "mw": 1, "reaction_kapp_change_fil": 1, "new": 1, "adj_trans_model2enz_model": [1, 3], "f": 1, "ptot": 1, "sigma": 1, "lowerbound": 1, "upperbound": 1, "json_output_fil": 1, "tansform": 1, "mode": 1, "concentr": 1, "constraintat": 1, "catalyz": 1, "gem": 1, "fraction": 1, "total": 1, "cell": 1, "approxim": 1, "averag": 1, "satur": 1, "constraint": 1, "calculate_f": [1, 3], "gene_abundance_fil": 1, "gene_mw_fil": 1, "gene_mw_colnam": 1, "gene_abundance_colnam": 1, "base": [1, 4, 5], "abund": 1, "molecular": [1, 5], "weight": 1, "column": 1, "calculate_f_v2": [1, 3], "taxonom_id": 1, "calculate_yield": [1, 3], "use_substr": 1, "substrate_nam": 1, "glc_concentr": 1, "correspond_rxn": 1, "yield_list": 1, "yield": 1, "glucos": 1, "correspond": 1, "pd": 1, "updat": 1, "change_enz_model_by_enz_usag": [1, 3], "enz_ratio": 1, "reaction_flux_fil": 1, "usag": 1, "calibr": 1, "ratio": [1, 4], "flux": 1, "reaction_enz_usage_fil": 1, "reaction_mw_outfil": 1, "change_reaction_kcat_by_databas": [1, 3], "combine_reactions_simles_sequ": [1, 3], "smilesdf": 1, "prodf": 1, "comdf_fil": 1, "protein_sequnc": 1, "metabolites_reactions_gpr": 1, "similesdf": 1, "inchkeydf": 1, "convert_bigg_met_to_inchikei": [1, 3], "metlist": 1, "inchikey_list_fil": 1, "inchikei": 1, "convert_inchikey_to_smil": [1, 3], "inchikey_list_smilesfil": 1, "smile": 1, "canonical_smil": 1, "convert_to_irrevers": [1, 3], "split": 1, "two": 1, "irrevers": 1, "These": 1, "proce": 1, "opposit": 1, "guarente": 1, "allow": 1, "posit": 1, "some": 1, "problem": 1, "modifi": 1, "place": 1, "directori": 1, "specifi": 1, "arg": 1, "create_trac": [1, 3], "x": [1, 4], "y": 1, "color": 1, "size": 1, "10": 1, "symbol": 1, "width": 1, "xaxi": 1, "yaxi": 1, "scatter": 1, "trace": 1, "plot": 1, "arrai": 1, "axi": 1, "marker": 1, "option": 1, "go": 1, "draw_3d_rba": [1, 3], "z_data": 1, "substrate_bound": 1, "o2_bound": 1, "obj_bound": 1, "phpp_output_fig_fil": 1, "gener": 1, "3d": 1, "analysi": 1, "rba": 1, "z": 1, "upper": 1, "o2": 1, "uptak": 1, "rate": 1, "growth": 1, "point": 1, "figur": 1, "fig": 1, "draw_cdf_fig": [1, 3], "data_cdf_data": 1, "output_fil": 1, "x_name": 1, "y_name": 1, "y_index": 1, "ntick": 1, "cumul": 1, "distribut": 1, "cdf": 1, "index": [1, 3], "tick": 1, "draw_overfolw_fig": [1, 3], "gemyield_list": 1, "ecgemyield_list": 1, "column_list": 1, "y_axis_loc_list": 1, "color_list": 1, "secrate_bound": 1, "trade_off_biomass_yield_figfil": 1, "show": 1, "trade": 1, "off": 1, "between": 1, "biomass": 1, "secret": 1, "locat": 1, "left": 1, "right": 1, "limit": 1, "plotli": 1, "draw_trade_off": [1, 3], "efficiency_pfba": 1, "trade_off_enzyme_efficiency_figfil": 1, "draw": 1, "relationship": 1, "effici": 1, "drawphpp": [1, 3], "gem_glc_o2_df": 1, "ecgem_glc_o2_df": 1, "surfac": 1, "phpp": 1, "oxygen": [1, 5], "interv": 1, "divid": 1, "extract_metabolite_id": [1, 3], "met_id": 1, "extract": [1, 5], "compart": 1, "generate_dlkcat_input": [1, 3], "metdf_nam": 1, "metdf_outfil": 1, "dlinputdf_fil": 1, "generate_random_color": [1, 3], "num_color": 1, "hexadecim": 1, "get_phpp_data": [1, 3], "model_typ": 1, "obj": 1, "outputfil": 1, "x_id": 1, "z_id": 1, "type": 1, "genom": 1, "scale": 1, "other": 1, "df1": 1, "get_currency_metabolit": [1, 3], "currenc": 1, "get_enz_foldchang": [1, 3], "ecmodel_fil": 1, "substrate_con": 1, "biomass_id": 1, "biomass_min": 1, "biomass_max": 1, "fc_threshold": 1, "fluxes_outfil": 1, "enzcost_diff_fil": 1, "fold": 1, "minimum": 1, "maximum": 1, "condit": 1, "cost": 1, "enzcost_diff": 1, "get_enz_model_use_enz_usage_by_eckcat": [1, 3], "get_enzyme_constraint_model": [1, 3], "json_model_fil": 1, "get_enzyme_constraint_model_perc": [1, 3], "percent": 1, "get_fluxes_detail_in_model": [1, 3], "model_pfba_solut": 1, "detail": 1, "metabol": 1, "seri": 1, "pfba": 1, "solut": 1, "model_pfba_solution_detail": 1, "get_gene_subunitdescript": [1, 3], "sub_description_path": 1, "subunit": 1, "descript": 1, "get_genes_and_gpr": [1, 3], "gene_outfil": 1, "gpr_outfil": 1, "gene_reaction_rul": 1, "network": 1, "gpr": 1, "get_met_bigg_id": [1, 3], "metdf_uniqu": 1, "uniqu": 1, "get_min_enzyme_cost": [1, 3], "dict_coeff": 1, "coeffient": 1, "get_model_protein_sequence_and_mass": [1, 3], "subbnumdf": 1, "prodf_fil": 1, "get_model_substrate_obj": [1, 3], "use_model": 1, "singl": [1, 5], "carbon": 1, "get_protein_mass_mapping_from_loc": [1, 3], "uniprot_data_fil": 1, "obtain": 1, "get_reaction_kcat_mw": [1, 3], "type_of_default_kcat_select": 1, "enzyme_unit_number_fil": 1, "add": 1, "proteom": 1, "accord": 1, "smoment": 1, "stoichiometr": 1, "represent": 1, "order": 1, "spreadsheet": 1, "supplement": 1, "creation": 1, "avail": 1, "max": 1, "null": 1, "unit": 1, "get_reaction_kcatmw_onestop": [1, 3], "work_fold": 1, "kcat_method": 1, "bigg_metabolites_fil": 1, "kcat_gap_fil": 1, "reaction_gap_fil": 1, "get_reaction_kcatmw_onestop_by_autopacmen": [1, 3], "autopacmen_fold": 1, "get_reaction_kcatmw_onestop_by_dlkcat": [1, 3], "dlkcat_fold": 1, "get_reaction_mw": [1, 3], "repres": 1, "get_subunit_numb": [1, 3], "gene_subnum_path": 1, "df": 1, "get_yield_cost_effici": [1, 3], "enz_model": 1, "glc_concentration_list": 1, "efficiency_fil": 1, "evalu": 1, "yield_cost_efficiency_df": 1, "isoenzyme_split": [1, 3], "isoenzym": 1, "mutipl": 1, "preprocess_metabolit": [1, 3], "currency_metabolit": 1, "preprocess": 1, "run_fseof": [1, 3], "fseof_fil": 1, "scan": 1, "feasibl": 1, "optim": 1, "fseof": 1, "fseofdf_f": 1, "feseof_gen": 1, "regul": 1, "split_substrate_to_match_gen": [1, 3], "metabolites_reactions_gpr_fil": 1, "match": 1, "trans_model2enz_json_model_split_isoenzym": [1, 3], "model_main": [2, 3], "autopacmen_funct": 3, "ecmpy_funct": 3, "get_ecgem_onestop": 3, "kcatpredict": [3, 4], "attention_cnn": [3, 4], "gnn": [3, 4], "tester": [3, 4], "save_ma": [3, 4], "save_model": [3, 4], "test": [3, 4], "trainer": [3, 4], "train": [3, 4], "load_pickl": [3, 4], "load_tensor": [3, 4, 5], "shuffle_dataset": [3, 4], "split_dataset": [3, 4], "prediction_for_input": 3, "predictor": [3, 5], "predict": [3, 5], "create_adjac": [3, 5], "create_atom": [3, 5], "create_ijbonddict": [3, 5], "dump_dictionari": [3, 5], "extract_fingerprint": [3, 5], "get_smil": [3, 5], "main": [3, 5], "split_sequ": [3, 5], "page": 3, "class": [4, 5], "devic": 4, "n_fingerprint": 4, "n_word": 4, "dim": 4, "layer_gnn": 4, "window": 4, "layer_cnn": 4, "layer_output": 4, "layer": 4, "attent": 4, "mechan": 4, "appli": 4, "last": 4, "cnn": 4, "defin": 4, "comput": 4, "everi": 4, "should": 4, "overridden": 4, "subclass": 4, "although": 4, "recip": 4, "pass": 4, "within": 4, "instanc": 4, "afterward": 4, "sinc": 4, "former": 4, "take": 4, "care": 4, "run": 4, "regist": 4, "hook": 4, "while": 4, "latter": 4, "silent": 4, "ignor": 4, "mae": 4, "dataset": 4, "file_nam": [4, 5], "dtype": [4, 5], "seed": 4, "mol": 5, "atom": 5, "hydrogen": 5, "consid": 5, "aromat": 5, "node": 5, "tupl": 5, "neighbor": 5, "bond": 5, "doubl": 5, "i_jbond_dict": 5, "radiu": 5, "r": 5, "subgraph": 5, "fingerprint": 5, "graph": 5, "weisfeil": 5, "lehman": 5, "algorithm": 5, "ngram": 5}, "objects": {"": [[0, 0, 0, "-", "AutoPACMEN_function"], [1, 0, 0, "-", "ECMpy_function"], [2, 0, 0, "-", "get_ecGEM_onestop"], [4, 0, 0, "-", "model"], [5, 0, 0, "-", "prediction_for_input"]], "AutoPACMEN_function": [[0, 1, 1, "", "create_combined_kcat_database"], [0, 1, 1, "", "create_file"], [0, 1, 1, "", "ensure_folder_existence"], [0, 1, 1, "", "get_ec_number_kcats_wildcard_search"], [0, 1, 1, "", "get_entrez_id_from_organism_full_name"], [0, 1, 1, "", "get_entrez_id_from_organism_full_name_batch"], [0, 1, 1, "", "get_files"], [0, 1, 1, "", "get_id_associated_kcats"], [0, 1, 1, "", "get_protein_mass_mapping"], [0, 1, 1, "", "get_protein_mass_mapping_with_sbml"], [0, 1, 1, "", "get_reactions_kcat_mapping"], [0, 1, 1, "", "get_taxonomy_from_organism_ncbi_id"], [0, 1, 1, "", "get_taxonomy_from_organism_ncbi_id_batch"], [0, 1, 1, "", "is_fitting_ec_numbers"], [0, 1, 1, "", "json_load"], [0, 1, 1, "", "json_write"], [0, 1, 1, "", "most_taxonomic_similar"], [0, 1, 1, "", "parse_bigg_metabolites_file"], [0, 1, 1, "", "parse_brenda_json_for_model"], [0, 1, 1, "", "parse_brenda_textfile"], [0, 1, 1, "", "parse_sabio_rk_for_eclist"], [0, 1, 1, "", "parse_sabio_rk_for_model"], [0, 1, 1, "", "parse_sabio_rk_for_model_with_sbml"], [0, 1, 1, "", "pickle_load"], [0, 1, 1, "", "pickle_write"], [0, 1, 1, "", "sabio_rk_query_get_csv_lines"], [0, 1, 1, "", "sabio_rk_query_with_query_dicts"], [0, 1, 1, "", "sabio_rk_query_with_string"], [0, 1, 1, "", "standardize_folder"]], "ECMpy_function": [[1, 1, 1, "", "DL_kcat_mw_calculation"], [1, 1, 1, "", "Determine_suitable_ecGEM"], [1, 1, 1, "", "GENENAME_2_ACC_from_uniprot"], [1, 1, 1, "", "adj_reaction_kcat_by_database"], [1, 1, 1, "", "adj_trans_model2enz_model"], [1, 1, 1, "", "calculate_f"], [1, 1, 1, "", "calculate_f_v2"], [1, 1, 1, "", "calculate_yield"], [1, 1, 1, "", "change_enz_model_by_enz_usage"], [1, 1, 1, "", "change_reaction_kcat_by_database"], [1, 1, 1, "", "combine_reactions_simles_sequence"], [1, 1, 1, "", "convert_bigg_met_to_inchikey"], [1, 1, 1, "", "convert_inchikey_to_smiles"], [1, 1, 1, "", "convert_to_irreversible"], [1, 1, 1, "", "create_file"], [1, 1, 1, "", "create_trace"], [1, 1, 1, "", "draw_3d_rbas"], [1, 1, 1, "", "draw_cdf_fig"], [1, 1, 1, "", "draw_overfolw_fig"], [1, 1, 1, "", "draw_trade_off"], [1, 1, 1, "", "drawphpp"], [1, 1, 1, "", "extract_metabolite_id"], [1, 1, 1, "", "generate_DLKCAT_input"], [1, 1, 1, "", "generate_random_colors"], [1, 1, 1, "", "get_PhPP_data"], [1, 1, 1, "", "get_currency_metabolites"], [1, 1, 1, "", "get_enz_foldchange"], [1, 1, 1, "", "get_enz_model_use_enz_usage_by_eckcat"], [1, 1, 1, "", "get_enzyme_constraint_model"], [1, 1, 1, "", "get_enzyme_constraint_model_percent"], [1, 1, 1, "", "get_fluxes_detail_in_model"], [1, 1, 1, "", "get_gene_subunitDescription"], [1, 1, 1, "", "get_genes_and_gpr"], [1, 1, 1, "", "get_met_bigg_id"], [1, 1, 1, "", "get_min_enzyme_cost"], [1, 1, 1, "", "get_model_protein_sequence_and_mass"], [1, 1, 1, "", "get_model_substrate_obj"], [1, 1, 1, "", "get_protein_mass_mapping_from_local"], [1, 1, 1, "", "get_reaction_kcat_mw"], [1, 1, 1, "", "get_reaction_kcatmw_onestop"], [1, 1, 1, "", "get_reaction_kcatmw_onestop_by_AutoPACMEN"], [1, 1, 1, "", "get_reaction_kcatmw_onestop_by_DLKcat"], [1, 1, 1, "", "get_reaction_mw"], [1, 1, 1, "", "get_subunit_number"], [1, 1, 1, "", "get_yield_cost_efficiency"], [1, 1, 1, "", "isoenzyme_split"], [1, 1, 1, "", "json_load"], [1, 1, 1, "", "json_write"], [1, 1, 1, "", "parse_sabio_rk_for_eclist"], [1, 1, 1, "", "preprocess_metabolites"], [1, 1, 1, "", "run_FSEOF"], [1, 1, 1, "", "split_substrate_to_match_gene"], [1, 1, 1, "", "standardize_folder"], [1, 1, 1, "", "trans_model2enz_json_model_split_isoenzyme"]], "get_ecGEM_onestop": [[2, 1, 1, "", "model_main"]], "model": [[4, 2, 1, "", "KcatPrediction"], [4, 2, 1, "", "Tester"], [4, 2, 1, "", "Trainer"], [4, 1, 1, "", "load_pickle"], [4, 1, 1, "", "load_tensor"], [4, 1, 1, "", "shuffle_dataset"], [4, 1, 1, "", "split_dataset"]], "model.KcatPrediction": [[4, 3, 1, "", "attention_cnn"], [4, 3, 1, "", "forward"], [4, 3, 1, "", "gnn"]], "model.Tester": [[4, 3, 1, "", "save_MAEs"], [4, 3, 1, "", "save_model"], [4, 3, 1, "", "test"]], "model.Trainer": [[4, 3, 1, "", "train"]], "prediction_for_input": [[5, 2, 1, "", "Predictor"], [5, 1, 1, "", "create_adjacency"], [5, 1, 1, "", "create_atoms"], [5, 1, 1, "", "create_ijbonddict"], [5, 1, 1, "", "dump_dictionary"], [5, 1, 1, "", "extract_fingerprints"], [5, 1, 1, "", "get_smiles"], [5, 1, 1, "", "load_tensor"], [5, 1, 1, "", "main"], [5, 1, 1, "", "split_sequence"]], "prediction_for_input.Predictor": [[5, 3, 1, "", "predict"]]}, "objtypes": {"0": "py:module", "1": "py:function", "2": "py:class", "3": "py:method"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "function", "Python function"], "2": ["py", "class", "Python class"], "3": ["py", "method", "Python method"]}, "titleterms": {"autopacmen_funct": 0, "argument": [0, 1], "output": [0, 1], "input": 0, "algorithm": [0, 1], "ecmpy_funct": 1, "return": 1, "get_ecgem_onestop": 2, "document": 3, "ecmpi": 3, "indic": 3, "tabl": 3, "model": 4, "prediction_for_input": 5}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.todo": 2, "sphinx": 57}, "alltitles": {"AutoPACMEN_function": [[0, "module-AutoPACMEN_function"]], "Arguments": [[0, "arguments"], [0, "id1"], [0, "id8"], [0, "id10"], [0, "id12"], [0, "id13"], [0, "id31"], [0, "id32"], [0, "id33"], [0, "id34"], [0, "id35"], [0, "id37"], [0, "id39"], [0, "id41"], [0, "id44"], [0, "id46"], [0, "id47"], [0, "id48"], [0, "id53"], [1, "arguments"], [1, "id1"], [1, "id2"], [1, "id3"], [1, "id4"], [1, "id5"], [1, "id6"], [1, "id8"], [1, "id9"], [1, "id10"], [1, "id11"], [1, "id12"], [1, "id14"], [1, "id15"], [1, "id17"], [1, "id18"], [1, "id22"], [1, "id23"], [1, "id24"], [1, "id25"], [1, "id27"], [1, "id29"], [1, "id31"]], "Argument": [[0, "argument"], [0, "id55"], [1, "argument"]], "Output": [[0, "output"], [0, "id9"], [0, "id11"], [0, "id30"], [0, "id36"], [0, "id38"], [0, "id40"], [0, "id42"], [0, "id45"], [0, "id50"], [0, "id52"], [0, "id54"], [1, "output"], [1, "id26"]], "Input": [[0, "input"], [0, "id49"], [0, "id51"]], "Algorithm": [[0, "algorithm"], [0, "id43"], [1, "algorithm"]], "ECMpy_function": [[1, "module-ECMpy_function"]], "Returns": [[1, "returns"], [1, "id7"], [1, "id13"], [1, "id16"], [1, "id28"], [1, "id30"]], "Arguments:": [[1, "id19"]], "Output:": [[1, "id20"], [1, "id21"]], "get_ecGEM_onestop": [[2, "module-get_ecGEM_onestop"]], "Documentation for ecmpy": [[3, "documentation-for-ecmpy"]], "Indices and tables": [[3, "indices-and-tables"]], "model": [[4, "module-model"]], "prediction_for_input": [[5, "module-prediction_for_input"]]}, "indexentries": {"autopacmen_function": [[0, "module-AutoPACMEN_function"]], "create_combined_kcat_database() (in module autopacmen_function)": [[0, "AutoPACMEN_function.create_combined_kcat_database"]], "create_file() (in module autopacmen_function)": [[0, "AutoPACMEN_function.create_file"]], "ensure_folder_existence() (in module autopacmen_function)": [[0, "AutoPACMEN_function.ensure_folder_existence"]], "get_ec_number_kcats_wildcard_search() (in module autopacmen_function)": [[0, "AutoPACMEN_function.get_ec_number_kcats_wildcard_search"]], "get_entrez_id_from_organism_full_name() (in module autopacmen_function)": [[0, "AutoPACMEN_function.get_entrez_id_from_organism_full_name"]], "get_entrez_id_from_organism_full_name_batch() (in module autopacmen_function)": [[0, "AutoPACMEN_function.get_entrez_id_from_organism_full_name_batch"]], "get_files() (in module autopacmen_function)": [[0, "AutoPACMEN_function.get_files"]], "get_id_associated_kcats() (in module autopacmen_function)": [[0, "AutoPACMEN_function.get_id_associated_kcats"]], "get_protein_mass_mapping() (in module autopacmen_function)": [[0, "AutoPACMEN_function.get_protein_mass_mapping"]], "get_protein_mass_mapping_with_sbml() (in module autopacmen_function)": [[0, "AutoPACMEN_function.get_protein_mass_mapping_with_sbml"]], "get_reactions_kcat_mapping() (in module autopacmen_function)": [[0, "AutoPACMEN_function.get_reactions_kcat_mapping"]], "get_taxonomy_from_organism_ncbi_id() (in module autopacmen_function)": [[0, "AutoPACMEN_function.get_taxonomy_from_organism_ncbi_id"]], "get_taxonomy_from_organism_ncbi_id_batch() (in module autopacmen_function)": [[0, "AutoPACMEN_function.get_taxonomy_from_organism_ncbi_id_batch"]], "is_fitting_ec_numbers() (in module autopacmen_function)": [[0, "AutoPACMEN_function.is_fitting_ec_numbers"]], "json_load() (in module autopacmen_function)": [[0, "AutoPACMEN_function.json_load"]], "json_write() (in module autopacmen_function)": [[0, "AutoPACMEN_function.json_write"]], "module": [[0, "module-AutoPACMEN_function"], [1, "module-ECMpy_function"], [2, "module-get_ecGEM_onestop"], [4, "module-model"], [5, "module-prediction_for_input"]], "most_taxonomic_similar() (in module autopacmen_function)": [[0, "AutoPACMEN_function.most_taxonomic_similar"]], "parse_bigg_metabolites_file() (in module autopacmen_function)": [[0, "AutoPACMEN_function.parse_bigg_metabolites_file"]], "parse_brenda_json_for_model() (in module autopacmen_function)": [[0, "AutoPACMEN_function.parse_brenda_json_for_model"]], "parse_brenda_textfile() (in module autopacmen_function)": [[0, "AutoPACMEN_function.parse_brenda_textfile"]], "parse_sabio_rk_for_eclist() (in module autopacmen_function)": [[0, "AutoPACMEN_function.parse_sabio_rk_for_eclist"]], "parse_sabio_rk_for_model() (in module autopacmen_function)": [[0, "AutoPACMEN_function.parse_sabio_rk_for_model"]], "parse_sabio_rk_for_model_with_sbml() (in module autopacmen_function)": [[0, "AutoPACMEN_function.parse_sabio_rk_for_model_with_sbml"]], "pickle_load() (in module autopacmen_function)": [[0, "AutoPACMEN_function.pickle_load"]], "pickle_write() (in module autopacmen_function)": [[0, "AutoPACMEN_function.pickle_write"]], "sabio_rk_query_get_csv_lines() (in module autopacmen_function)": [[0, "AutoPACMEN_function.sabio_rk_query_get_csv_lines"]], "sabio_rk_query_with_query_dicts() (in module autopacmen_function)": [[0, "AutoPACMEN_function.sabio_rk_query_with_query_dicts"]], "sabio_rk_query_with_string() (in module autopacmen_function)": [[0, "AutoPACMEN_function.sabio_rk_query_with_string"]], "standardize_folder() (in module autopacmen_function)": [[0, "AutoPACMEN_function.standardize_folder"]], "dl_kcat_mw_calculation() (in module ecmpy_function)": [[1, "ECMpy_function.DL_kcat_mw_calculation"]], "determine_suitable_ecgem() (in module ecmpy_function)": [[1, "ECMpy_function.Determine_suitable_ecGEM"]], "ecmpy_function": [[1, "module-ECMpy_function"]], "genename_2_acc_from_uniprot() (in module ecmpy_function)": [[1, "ECMpy_function.GENENAME_2_ACC_from_uniprot"]], "adj_reaction_kcat_by_database() (in module ecmpy_function)": [[1, "ECMpy_function.adj_reaction_kcat_by_database"]], "adj_trans_model2enz_model() (in module ecmpy_function)": [[1, "ECMpy_function.adj_trans_model2enz_model"]], "calculate_f() (in module ecmpy_function)": [[1, "ECMpy_function.calculate_f"]], "calculate_f_v2() (in module ecmpy_function)": [[1, "ECMpy_function.calculate_f_v2"]], "calculate_yield() (in module ecmpy_function)": [[1, "ECMpy_function.calculate_yield"]], "change_enz_model_by_enz_usage() (in module ecmpy_function)": [[1, "ECMpy_function.change_enz_model_by_enz_usage"]], "change_reaction_kcat_by_database() (in module ecmpy_function)": [[1, "ECMpy_function.change_reaction_kcat_by_database"]], "combine_reactions_simles_sequence() (in module ecmpy_function)": [[1, "ECMpy_function.combine_reactions_simles_sequence"]], "convert_bigg_met_to_inchikey() (in module ecmpy_function)": [[1, "ECMpy_function.convert_bigg_met_to_inchikey"]], "convert_inchikey_to_smiles() (in module ecmpy_function)": [[1, "ECMpy_function.convert_inchikey_to_smiles"]], "convert_to_irreversible() (in module ecmpy_function)": [[1, "ECMpy_function.convert_to_irreversible"]], "create_file() (in module ecmpy_function)": [[1, "ECMpy_function.create_file"]], "create_trace() (in module ecmpy_function)": [[1, "ECMpy_function.create_trace"]], "draw_3d_rbas() (in module ecmpy_function)": [[1, "ECMpy_function.draw_3d_rbas"]], "draw_cdf_fig() (in module ecmpy_function)": [[1, "ECMpy_function.draw_cdf_fig"]], "draw_overfolw_fig() (in module ecmpy_function)": [[1, "ECMpy_function.draw_overfolw_fig"]], "draw_trade_off() (in module ecmpy_function)": [[1, "ECMpy_function.draw_trade_off"]], "drawphpp() (in module ecmpy_function)": [[1, "ECMpy_function.drawphpp"]], "extract_metabolite_id() (in module ecmpy_function)": [[1, "ECMpy_function.extract_metabolite_id"]], "generate_dlkcat_input() (in module ecmpy_function)": [[1, "ECMpy_function.generate_DLKCAT_input"]], "generate_random_colors() (in module ecmpy_function)": [[1, "ECMpy_function.generate_random_colors"]], "get_phpp_data() (in module ecmpy_function)": [[1, "ECMpy_function.get_PhPP_data"]], "get_currency_metabolites() (in module ecmpy_function)": [[1, "ECMpy_function.get_currency_metabolites"]], "get_enz_foldchange() (in module ecmpy_function)": [[1, "ECMpy_function.get_enz_foldchange"]], "get_enz_model_use_enz_usage_by_eckcat() (in module ecmpy_function)": [[1, "ECMpy_function.get_enz_model_use_enz_usage_by_eckcat"]], "get_enzyme_constraint_model() (in module ecmpy_function)": [[1, "ECMpy_function.get_enzyme_constraint_model"]], "get_enzyme_constraint_model_percent() (in module ecmpy_function)": [[1, "ECMpy_function.get_enzyme_constraint_model_percent"]], "get_fluxes_detail_in_model() (in module ecmpy_function)": [[1, "ECMpy_function.get_fluxes_detail_in_model"]], "get_gene_subunitdescription() (in module ecmpy_function)": [[1, "ECMpy_function.get_gene_subunitDescription"]], "get_genes_and_gpr() (in module ecmpy_function)": [[1, "ECMpy_function.get_genes_and_gpr"]], "get_met_bigg_id() (in module ecmpy_function)": [[1, "ECMpy_function.get_met_bigg_id"]], "get_min_enzyme_cost() (in module ecmpy_function)": [[1, "ECMpy_function.get_min_enzyme_cost"]], "get_model_protein_sequence_and_mass() (in module ecmpy_function)": [[1, "ECMpy_function.get_model_protein_sequence_and_mass"]], "get_model_substrate_obj() (in module ecmpy_function)": [[1, "ECMpy_function.get_model_substrate_obj"]], "get_protein_mass_mapping_from_local() (in module ecmpy_function)": [[1, "ECMpy_function.get_protein_mass_mapping_from_local"]], "get_reaction_kcat_mw() (in module ecmpy_function)": [[1, "ECMpy_function.get_reaction_kcat_mw"]], "get_reaction_kcatmw_onestop() (in module ecmpy_function)": [[1, "ECMpy_function.get_reaction_kcatmw_onestop"]], "get_reaction_kcatmw_onestop_by_autopacmen() (in module ecmpy_function)": [[1, "ECMpy_function.get_reaction_kcatmw_onestop_by_AutoPACMEN"]], "get_reaction_kcatmw_onestop_by_dlkcat() (in module ecmpy_function)": [[1, "ECMpy_function.get_reaction_kcatmw_onestop_by_DLKcat"]], "get_reaction_mw() (in module ecmpy_function)": [[1, "ECMpy_function.get_reaction_mw"]], "get_subunit_number() (in module ecmpy_function)": [[1, "ECMpy_function.get_subunit_number"]], "get_yield_cost_efficiency() (in module ecmpy_function)": [[1, "ECMpy_function.get_yield_cost_efficiency"]], "isoenzyme_split() (in module ecmpy_function)": [[1, "ECMpy_function.isoenzyme_split"]], "json_load() (in module ecmpy_function)": [[1, "ECMpy_function.json_load"]], "json_write() (in module ecmpy_function)": [[1, "ECMpy_function.json_write"]], "parse_sabio_rk_for_eclist() (in module ecmpy_function)": [[1, "ECMpy_function.parse_sabio_rk_for_eclist"]], "preprocess_metabolites() (in module ecmpy_function)": [[1, "ECMpy_function.preprocess_metabolites"]], "run_fseof() (in module ecmpy_function)": [[1, "ECMpy_function.run_FSEOF"]], "split_substrate_to_match_gene() (in module ecmpy_function)": [[1, "ECMpy_function.split_substrate_to_match_gene"]], "standardize_folder() (in module ecmpy_function)": [[1, "ECMpy_function.standardize_folder"]], "trans_model2enz_json_model_split_isoenzyme() (in module ecmpy_function)": [[1, "ECMpy_function.trans_model2enz_json_model_split_isoenzyme"]], "get_ecgem_onestop": [[2, "module-get_ecGEM_onestop"]], "model_main() (in module get_ecgem_onestop)": [[2, "get_ecGEM_onestop.model_main"]], "kcatprediction (class in model)": [[4, "model.KcatPrediction"]], "tester (class in model)": [[4, "model.Tester"]], "trainer (class in model)": [[4, "model.Trainer"]], "attention_cnn() (model.kcatprediction method)": [[4, "model.KcatPrediction.attention_cnn"]], "forward() (model.kcatprediction method)": [[4, "model.KcatPrediction.forward"]], "gnn() (model.kcatprediction method)": [[4, "model.KcatPrediction.gnn"]], "load_pickle() (in module model)": [[4, "model.load_pickle"]], "load_tensor() (in module model)": [[4, "model.load_tensor"]], "model": [[4, "module-model"]], "save_maes() (model.tester method)": [[4, "model.Tester.save_MAEs"]], "save_model() (model.tester method)": [[4, "model.Tester.save_model"]], "shuffle_dataset() (in module model)": [[4, "model.shuffle_dataset"]], "split_dataset() (in module model)": [[4, "model.split_dataset"]], "test() (model.tester method)": [[4, "model.Tester.test"]], "train() (model.trainer method)": [[4, "model.Trainer.train"]], "predictor (class in prediction_for_input)": [[5, "prediction_for_input.Predictor"]], "create_adjacency() (in module prediction_for_input)": [[5, "prediction_for_input.create_adjacency"]], "create_atoms() (in module prediction_for_input)": [[5, "prediction_for_input.create_atoms"]], "create_ijbonddict() (in module prediction_for_input)": [[5, "prediction_for_input.create_ijbonddict"]], "dump_dictionary() (in module prediction_for_input)": [[5, "prediction_for_input.dump_dictionary"]], "extract_fingerprints() (in module prediction_for_input)": [[5, "prediction_for_input.extract_fingerprints"]], "get_smiles() (in module prediction_for_input)": [[5, "prediction_for_input.get_smiles"]], "load_tensor() (in module prediction_for_input)": [[5, "prediction_for_input.load_tensor"]], "main() (in module prediction_for_input)": [[5, "prediction_for_input.main"]], "predict() (prediction_for_input.predictor method)": [[5, "prediction_for_input.Predictor.predict"]], "prediction_for_input": [[5, "module-prediction_for_input"]], "split_sequence() (in module prediction_for_input)": [[5, "prediction_for_input.split_sequence"]]}})